`atan2()` - New CSS functional notation (#20592)
mdn/content: files/en-us/web/css/atan2/index.md

---
title: atan2()
slug: Web/CSS/atan2
tags:
  - CSS
  - CSS Function
  - Function
  - Math
  - Reference
  - Web
  - atan2
  - Experimental
browser-compat: css.types.atan2
spec-urls: https://drafts.csswg.org/css-values/#trig-funcs
---
{{CSSRef}}{{SeeCompatTable}}

> The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of a number between `-infinity` and `infinity`. The function contains a calculation that returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.

Reviewer: ```suggestion
The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of two values between `-infinity` and `infinity`. The function accepts two arguments and returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.
```

## Syntax

```css
/* Single <number> values */
transform: rotate(atan2(90, 15));
transform: rotate(atan2(15, 90));

/* Other values */
transform: rotate(atan2(pi, 45));
transform: rotate(atan2(e, 30));
```

> ### Parameter

Reviewer: ```suggestion
### Parameters
```

> The `atan2(y, x)` function accepts two comma-separated value as its parameters.

Reviewer: I would mention at this point what the values that `x` and `y` can be, explaining that they can be as described below, but both values must be of the same type.

```suggestion
The `atan2(y, x)` function accepts two comma-separated values as its parameters.
The values can be a {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}, but both must be the same type.
```

Reviewer: I would also add:

>Given two points `x` and `y`, `atan2(y, x) calculates the angle between the positive x axis and the ray from the origin to the point (x, y).

---
title: atan2()
slug: Web/CSS/atan2
tags:
  - CSS
  - CSS Function
  - Function
  - Math
  - Reference
  - Web
  - atan2
  - Experimental
browser-compat: css.types.atan2
spec-urls: https://drafts.csswg.org/css-values/#trig-funcs
---
{{CSSRef}}{{SeeCompatTable}}

The **`atan2()`** [CSS](/en-US/docs/Web/CSS) [function](/en-US/docs/Web/CSS/CSS_Functions) is a trigonometric function that returns the inverse tangent of two values between `-infinity` and `infinity`. The function accepts two arguments and returns the number of radians representing an {{cssxref("&lt;angle&gt;")}} between `-180deg` and `180deg`.

## Syntax

```css
/* Single <number> values */
transform: rotate(atan2(90, 15));
transform: rotate(atan2(15, 90));

> 

Reviewer: Can we add some examples for `<dimension>` and `<percentage>` also?

/* Other values */
transform: rotate(atan2(pi, 45));
transform: rotate(atan2(e, 30));
```

### Parameter

The `atan2(y, x)` function accepts two comma-separated values as its parameters.
The values can be a {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}, but both must be the same type.

- `y`
  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}.

- `x`
  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage>&gt;")}}.

### Return value

The function returns the {{cssxref("&lt;angle&gt;")}} between the positive X-axis and the point (x,y).

> 

Reviewer: Shall we add some examples like the other pages? There's a pen here that shows the different types:

https://codepen.io/bsmth/pen/gOzwrjo

Author: I'm using Firefox, with the trig functions flag enabled, but the `atan2()` doesn't rotate the boxes...

Reviewer: You might need a later version, I'm using `106.0a1 (2022-09-10)` from nightly. It also works in Safari (`Version 15.6.1 (17613.3.9.1.16)` for me.

Reviewer: It's also working on the [105.0b9 Beta](https://www.mozilla.org/en-US/firefox/channel/desktop/#beta) but it's behind the trig pref (`layout.css.trig.enabled`) like you mentioned.

## Syntax

```css
/* Single <number> values */
transform: rotate(atan2(3, 2));

/* Single <dimension> values */
transform: rotate(atan2(1rem, -0.5rem));

/* Single <percentage> values */
transform: rotate(atan2(20%, -30%));

/* Other values */
transform: rotate(atan2(pi, 45));
transform: rotate(atan2(e, 30));
```

### Parameters

The `atan2(y, x)` function accepts two comma-separated values as its parameters. The values can be a {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage&gt;")}}, but both must be the same type.

- `y`
  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage&gt;")}}.

- `x`
  - : The y coordinate of the point. A {{cssxref("&lt;number&gt;")}}, {{cssxref("&lt;dimension&gt;")}} or {{cssxref("&lt;percentage&gt;")}}.

### Return value

Given two points `x` and `y`, the function `atan2(y, x)` calculates and returns the {{cssxref("&lt;angle&gt;")}} between the positive `x` axis and the ray from the origin to the point `(x, y)`.

### Formal syntax

{{CSSSyntax}}

> 

Reviewer: ```suggestion
## Example

The `atan2()` function can be used to {{cssxref("transform-function/rotate", "rotate")}} elements as it return an {{cssxref("&lt;angle&gt;")}}.

```html hidden
<div class="box box-1"></div>
<div class="box box-2"></div>
<div class="box box-3"></div>
<div class="box box-4"></div>
<div class="box box-5"></div>
```

```css hidden
body {
  height: 100vh;
  display: flex;
  justify-content: center;
  align-items: center;
  gap: 50px;
}
```

```css
div.box {
  width: 100px;
  height: 100px;
  background: linear-gradient(orange, red);
}
div.box-1 {
  transform: rotate(atan2(3, 2));
}
div.box-2 {
  transform: rotate(atan2(3%, -2%));
}
div.box-3 {
  transform: rotate(atan2(-1, 0.5));
}
div.box-4 {
  transform: rotate(atan2(1, 0.5));
}
div.box-5 {
  transform: rotate(atan2(1rem, -0.5rem));
}
```

{{EmbedLiveSample('example', '100%', '200px')}}

```
